src/ImageSharp/Processing/Processors/Convolution/Convolution2DProcessor{TPixel}.cs

```diff
@@ -60,102 +57,39 @@ internal class Convolution2DProcessor<TPixel> : ImageProcessor<TPixel>
         /// <inheritdoc/>
         protected override void OnFrameApply(ImageFrame<TPixel> source)
         {
-            using Buffer2D<TPixel> targetPixels = this.Configuration.MemoryAllocator.Allocate2D<TPixel>(source.Width, source.Height);
+            MemoryAllocator allocator = this.Configuration.MemoryAllocator;
+            using Buffer2D<TPixel> targetPixels = allocator.Allocate2D<TPixel>(source.Width, source.Height);
 
             source.CopyTo(targetPixels);
 
             var interest = Rectangle.Intersect(this.SourceRectangle, source.Bounds());
-            var operation = new RowOperation(interest, targetPixels, source.PixelBuffer, this.KernelY, this.KernelX, this.Configuration, this.PreserveAlpha);
 
-            ParallelRowIterator.IterateRows<RowOperation, Vector4>(
-                this.Configuration,
-                interest,
-                in operation);
+            // We use a rectangle 3x the interest width to allocate a buffer big enough
+            // for source and target bulk pixel conversion.
+            var operationBounds = new Rectangle(interest.X, interest.Y, interest.Width * 3, interest.Height);
```

Comment:
I'm a bit confused about the temporary memory usage of this solution, as I remember you being a bit worried about this back when I was first working on the bokeh blur processor. Wouldn't this mean we're now allocating a `3 x [image size]` buffer every time now? As in, if I'm processing a `1920x1080` image, this would allocate a `5760 x 1080` temporary buffer, correct? Is that ok? At 4K image that'd be a buffer of 24 million pixels 🤔

Looking at the rest of the code I think I get why you're doing this (as it allows you to access pixels on the Y access in row major order), but I'd be curious about a diff in max peak memory usage compared to `master`, if you have made such a benchmark? If nothing else, I think it'd be an interesting bit of into to include 🙂

Reply:
We're actually only allocating a single of `Span<Vector4>` of length `3x interest width` per parallel region and only for 2D convolution. For 1D or 2Pass it's  `1x interest width`.

Reply:
Oooh I see, yeah that makes sense. Perfect then! 🚀